annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
         assert len(result[0]) == 2
         assert result[1] == expected_result[1]
         assert result[2] == expected_result[2]
+
+
+def test_create_tasks(mock_session: Mock):
+    tasks = [
+        {"user_id": 1, "file_id": 123, "pages": {1, 2}},
+        {"user_id": 2, "file_id": 456, "pages": {3}},
+    ]
+    job_id = 1
+    expected_user_ids = {1, 2}
+    with patch(
+        "annotation.tasks.services.update_files"
+    ) as mock_update_files, patch(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+
+        create_tasks(mock_session, tasks, job_id)
+
+        mock_session.bulk_insert_mappings.assert_called_once()
+        mock_update_files.assert_called_once_with(mock_session, tasks, job_id)
+        mock_update_user_overall_load.assert_has_calls(
+            [call(mock_session, user_id) for user_id in expected_user_ids],
+            any_order=True,
+        )
+
+
+def test_create_tasks_with_empty_tasks(mock_session: Mock):
+    tasks = []
+    job_id = 1
+    with patch(
+        "annotation.tasks.services.update_files"
+    ) as mock_update_files, patch(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+
+        create_tasks(mock_session, tasks, job_id)
+
+        mock_session.bulk_insert_mappings.assert_called_once()
+        mock_update_files.assert_called_once_with(mock_session, tasks, job_id)
+        mock_update_user_overall_load.assert_not_called()
+
+
+def create_task(status: TaskStatusEnumSchema) -> ManualAnnotationTask:
+    task = MagicMock(spec=ManualAnnotationTask)
+    task.status = status
+    task.id = 1
+    return task
+
+
+def test_update_task_status_ready(mock_session: Mock):
+    task = create_task(TaskStatusEnumSchema.ready)
+
+    update_task_status(mock_session, task)
+
+    assert task.status == TaskStatusEnumSchema.in_progress
+    mock_session.add.assert_called_once_with(task)
+    mock_session.commit.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("status", "expected_message"),
+    (
+        (TaskStatusEnumSchema.pending, "Job is not started yet"),
+        (TaskStatusEnumSchema.finished, "Task is already finished"),
+    ),
+)
+def test_update_task_status_error(
+    mock_session: Mock, status: TaskStatusEnumSchema, expected_message: str
+):
+    task = create_task(status)
+    with pytest.raises(FieldConstraintError, match=f".*{expected_message}.*"):
+        update_task_status(mock_session, task)
+
+
+def test_finish_validation_task(mock_session: MagicMock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.with_for_update.return_value = mock_query
+    mock_query.update.return_value = None
+
+    finish_validation_task(mock_session, mock_task)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+
```

Comment:
Use blank lines wisely -  it is good to visually separate logical blocks, but not that often - looks unpleasant. Especially to separate a single line of code